internal/olm/client/status.go

```diff
@@ -71,13 +79,49 @@ func (c Client) GetObjectsStatus(ctx context.Context, objs ...runtime.Object) St
 	return Status{Resources: rss}
 }
 
-func (s Status) HasExistingResources() bool {
+func (s Status) HasInstalledResources() (bool, error) {
+	crdKindSet, err := s.getCRDKindSet()
+	if err != nil {
+		return false, fmt.Errorf("error getting set of CRD kinds in resources: %w", err)
+	}
 	for _, r := range s.Resources {
 		if r.Resource != nil {
-			return true
+			return true, nil
+		} else if r.Error != nil && !apierrors.IsNotFound(r.Error) {
+			nkmerr := &meta.NoKindMatchError{}
+			if !errors.As(r.Error, &nkmerr) || !crdKindSet.Has(r.GVK.Kind) {
+				return false, r.Error
+			}
 		}
 	}
-	return false
+	return false, nil
+}
+
+func (s Status) getCRDKindSet() (sets.String, error) {
```

Comment:
```suggestion
// Resources will return a unique set of Kinds 
func (s Status) Resources() (sets.String, error) {
```

Should not make more sense it be in the Resources? It shows not specific for OLM.

Reply:
CRD's are not specific to OLM. This just retrieves any CRD kinds in `s`.

Reply:
Agreed. Because of this, I do not think that this impl should be in OLM. :-)

Reply:
I would agree, but the only code that needs the `Status` type is OLM code. If we find another use for it, we should move it.